multy_core/src/bitcoin/bitcoin_stream.cpp

```diff
@@ -0,0 +1,223 @@
+    {
+        *stream << static_cast<uint8_t>(254);
+        *stream << static_cast<uint32_t>(size);
+    }
+    else
+    {
+        *stream << static_cast<uint8_t>(255);
+        *stream << size;
+    }
+}
+
+template <typename T>
+BitcoinStream& write_as_data(const T& data, BitcoinStream& stream)
+{
+    return stream.write_data(
+            reinterpret_cast<const uint8_t*>(&data), sizeof(data));
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, uint8_t data)
+{
+    return write_as_data(data, stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, OP_CODE data)
+{
+    return write_as_data(data, stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, uint16_t data)
+{
+    return write_as_data(htole16(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, uint32_t data)
+{
+    return write_as_data(htole32(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, uint64_t data)
+{
+    return write_as_data(htole64(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, int8_t data)
+{
+    return write_as_data(data, stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, int16_t data)
+{
+    return write_as_data(htole16(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, int32_t data)
+{
+    return write_as_data(htole32(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, int64_t data)
+{
+    return write_as_data(htole64(data), stream);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, const BigInt& data)
+{
+    // Any BigInt we are going to serialize to transaction is non-negative.
+    return stream << data.get_value_as_uint64();
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, const BinaryData& data)
+{
+    return stream.write_data(
+            reinterpret_cast<const uint8_t*>(data.data), data.len);
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, const PublicKey& key)
+{
+    const BinaryData& key_data = key.get_content();
+    return stream << as_compact_size(key_data.len) << key_data;
+}
+
+ReversedBinaryData reverse(const BinaryData& data)
+{
+    return ReversedBinaryData{data};
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, const ReversedBinaryData& data)
+{
+    for (int i = data.data.len - 1; i >= 0; --i)
+    {
+        stream << data.data.data[i];
+    }
+    return stream;
+}
+
+BitcoinStream& operator<<(BitcoinStream& stream, const BitcoinTransactionSource& source)
+{
+
+
+    stream << reverse(**source.prev_transaction_hash);
+    stream << source.prev_transaction_out_index;
+
+    if (source.segwit == 0)
+    {
+        if (source.script_signature)
+        {
+            stream << as_compact_size(source.script_signature->len);
+            stream << *source.script_signature;
+        }
+        else
+        {
+            stream << as_compact_size(0);
+        }
+    }
+    else
+    {
+        const auto& public_key = source.private_key->make_public_key();
```

Comment:
Looks like an incomplete attempt to serialize a segwit source. And since we now have a separate implementation of SegWit source and proper serialization, it is time to remove this nonsense.